src/js/parser/formula.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * @author andreas.karner@student.tugraz.at
+ * @description Catblocks formular class for parsing catroid programs
+ */
+
+export default class Formula {
+  constructor() {
+    this.value = '';
+    this.operator = '';
+    this.left = null;
+    this.right = null;
+  }
+
+  setLeft(leftBlock) {
+    if (this.left === null) this.left = leftBlock;
+    else this.left.setLeft(leftBlock);
+  }
+
+  setRight(rightBlock) {
+    if (this.right === null) this.right = rightBlock;
+    else this.right.setRight(rightBlock);
+  }
+
+  static getAllLayouts() {
+    return {
+      'BRACKET': '(%l%r)',
+      'USER_LIST': '*%v*',
+      'SIN': '%v(%l)',
+      'COS': '%v(%l)',
+      'TAN': '%v(%l)',
+      'LN': '%v(%l)',
+      'LOG': '%v(%l)',
+      'ABS': '%v(%l)',
+      'ROUND': '%v(%l)',
+      'ARCSIN': '%v(%l)',
+      'ARCCOS': '%v(%l)',
+      'ARCTAN': '%v(%l)',
+      'FLOOR': '%v(%l)',
+      'CEIL': '%v(%l)',
+      'EXP': '%v(%l)',
+      'SQRT': '%v(%l)',
+      'MULTI_FINGER_X': '%v(%l)',
+      'MULTI_FINGER_Y': '%v(%l)',
+      'MULTI_FINGER_TOUCHED': '%v(%l)',
+      'ARCTAN2': '%v(%l, %r)',
+      'POWER': '%v(%l, %r)',
+      'MOD': '%v(%l, %r)',
+      'RAND': '%v(%l, %r)',
+      'MAX': '%v(%l, %r)',
+      'MIN': '%v(%l, %r)',
+      'LENGTH': '%v(%l%)',
+      'LETTER': '%v(%l, %r)',
+      'JOIN': '%v(%l, %r)',
+      'REGEX': '%v(%l, %r)',
+      'CONTAINS': '%v(%l, %r)',
+      'NUMBER_OF_ITEMS': '%v(%l)',
+      'LIST_ITEM': '%v(%l, %r)',
+      'DEFAULT': '%l %v %r'
+    };
```

Comment:
Formula operator output layout. %v get replaced with the operator value, %l with the left value and right value replaces %r. With this black magic, we can easily change the way how to render the formula.